src/object/class/internal.rs

```diff
@@ -65,7 +72,12 @@ impl Callback<()> for ConstructorCallCallback {
                 let data = info.data();
                 let kernel: fn(CallContext<JsValue>) -> JsResult<JsValue> =
                     mem::transmute(neon_runtime::class::get_call_kernel(data.to_raw()));
-                if let Ok(value) = convert_panics(|| { kernel(cx) }) {
+                #[cfg(feature = "napi-runtime")]
```

Comment:
We can avoid a lot of code duplication in the next couple chunks by changing the signature of `convert_panics` in `"legacy-runtime"` to match the same signature as `convert_panics` in `"napi-runtime"`. Then we can eliminate a lot of the `#[cfg(...)]` conditional-compilation branches.